kong/clustering/utils.lua

```diff
@@ -50,7 +51,7 @@ local function parse_proxy_url(proxy_server)
     end
 
     if parsed.user and parsed.password then
-      ret.proxy_authorization = "Basic " .. encode_base64(parsed.user  .. ":" .. parsed.password)
+      ret.proxy_authorization = "Basic " .. encode_base64(unescape_uri(parsed.user)  .. ":" .. unescape_uri(parsed.password))
```

Comment:
Should we also unescape the other parts of the url, for instance, the host?

Reply:
I've actually flashed this issue in my head before, and I'm guessing there aren't many cases where the `host` field uses special characters. But just in case, I'll deal with the situation now.